### PR TITLE
[DOC] What's Here for Net:HTTP

### PR DESCRIPTION
**Pull request by BurdetteLamar**
This is off-page b/c its headers collide with those on the main page.  Marking this as draft until that's agreed upon.  Will also need to add link from main page.

### CODE REVIEW

File: doc/net-http/whats_here.rdoc [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+- {:max_retries}[../../Net/HTTP.html#attribute-i-max_retries]:
+  Returns the maximum retries.
+- {:open_timeout}[../../Net/HTTP.html#attribute-i-open_timeout]:
+  Sets or returns the open timeout.
+- {:read_timeout}[../../Net/HTTP.html#attribute-i-read_timeout]:
+  Returns the open timeout.
+- {:ssl_timeout}[../../Net/HTTP.html#attribute-i-ssl_timeout]:
+  Returns the ssl timeout.
+- {:write_timeout}[../../Net/HTTP.html#attribute-i-write_timeout]:
+  Returns the write timeout.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/net-http/whats_here.rdoc:56]: I've never linked to an attribute before, and assumed (for no reason) that the rdoc:ref would not work.  I'll be fixing these in next push.

File: doc/net-http/whats_here.rdoc [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+- {#continue_timeout=}[rdoc-ref:Net::HTTP#continue_timeout=]:
+  Sets the continue timeout seconds.
+- {#max_retries=}[rdoc-ref:Net::HTTP#max_retries=]:
+  Sets the maximum retries.
+- {#read_timeout=}[rdoc-ref:Net::HTTP#read_timeout=]:
+  Sets the read timeout.
+- {write_timeout=}[rdoc-ref:Net::HTTP#write_timeout=]:
+  Sets the write timeout.
+
+Attributes:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/net-http/whats_here.rdoc:41]: But an attr_accessor is the exact same as two methods.  So, show them both as simply methods?

[the remark this answers is not in the document]
Comment (peterzhu2118) [reply doc/net-http/whats_here.rdoc:41]: Yeah I think that makes more sense.

File: doc/net-http/whats_here.rdoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/net-http/whats_here.rdoc]: Fixed.

File: doc/net-http/whats_here.rdoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/net-http/whats_here.rdoc]: Fixed.

**Comment by BurdetteLamar** [comment]
Still missing a link to this from the main page.  Where in the page should that be, do you think?

**Comment by BurdetteLamar** [comment]
> I think this would make the most sense to live in the documentation for `Net::HTTP`. WDYT?

I'd love that.  But there are colliding headings.  Those in the WH would have to be made less straightforward.  Worth it?

Also, regardless of whether a link or the full text, where in the page?

**Comment by BurdetteLamar** [comment]
> I'm not sure what the best solution to solving the heading collision is. I think the best way would be to embed this at the end of the `Net::HTTP` documentation rather than a link because a link introduces another level of indirection that makes it more difficult for the user to read and search.

Done.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, ready for you.

File: lib/net/http.rb [hunk 3]
```diff
@@ -454,6 +454,8 @@ class HTTPHeaderSyntaxError < StandardError; end
   # - Any other value:
   #   leaves the body and header unchanged.
   #
+  # :include: doc/net-http/whats_here.rdoc
```
Comment (peterzhu2118) [inline lib/net/http.rb:457]: Why do we do an include here instead of just pasting the contents?

Comment (BurdetteLamar) [reply lib/net/http.rb]: Done.

**Review by peterzhu2118 — APPROVED** [review]

**Comment by hsbt** [comment]
@BurdetteLamar This pull-request contains non-ascii character. see https://github.com/ruby/net-http/commit/d22ca54904deb1df295785d91a97390baa8a1875

**Comment by BurdetteLamar** [comment]
> @BurdetteLamar This pull-request contains non-ascii character. see [d22ca54](https://github.com/ruby/net-http/commit/d22ca54904deb1df295785d91a97390baa8a1875)

Thanks, @hsbt.

**Pull request merged**